src/dataExplorer/components/resources/index.ts

```diff
@@ -0,0 +1,38 @@
+import {ReactNode} from 'react'
+import {ResourceType} from 'src/types/resources'
+
+export interface ResourceConnectedQuery<T> {
+  type: ResourceType
+  flux: string
+  data: T
+}
+
+export interface ResourceRegistration {
+  type: ResourceType
+  disabled?: boolean
+  editor: ReactNode
+  init: (...args: string[]) => Promise<ResourceConnectedQuery<any>>
+  persist: (
+    query: ResourceConnectedQuery<any>
+  ) => Promise<ResourceConnectedQuery<any>>
+}
+
+interface Resources {
+  [type: string]: ResourceRegistration
+}
+
+export const RESOURCES: Resources = {}
+  // eslint-disable-next-line no-extra-semi
+;[].forEach(mod => {
```

Comment:
Is the `forEach` function intentionally applied on an empty array as a placeholder for the future code?

Reply:
YES